server/fly.io/Cargo.toml

```
@@ -0,0 +1,13 @@
+[package]
+name = "cargo-quickinstall-stats-server"
+version = "0.0.0"
+authors = ["David Laban <alsuren@gmail.com>"]
+# based on https://github.com/fly-apps/hello-rust by Jerome Gravel-Niquet <jeromegn@gmail.com>
+edition = "2021"
+
+# Make a new workspace so that we get our own Cargo.lock and target dir for the docker build. 
+[workspace]
+
```

Using the following profile should reduce binary size and improve performance:

```suggestion

[profile.release]
lto = true
codegen-units = 1
panic = "abort"
strip = "symbols"

```

---

If compilation time is not a concern, then you can also set:

```
lto = true # Use fat lto instead of thin lto, which provides better optimization
codegen-units = 1
```

---

This server is definitely not going to be performance sensitive. The nodejs version was plenty fast enough to handle 10 requests per hour or whatever, and the bottleneck will always be the database.

If anything, adding RUST_BACKTRACE=1 would be more helpful than panic=abort for this kind of thing 

Fly.io free tier doesn't even seem to put things to sleep when idle (unless I'm misreading something)

---

Got it, though I still think that setting `lto = "thin"` on release is reasonable.